runtime/reconcilers/project_parser.go

```diff
@@ -196,7 +196,11 @@ func (r *ProjectParserReconciler) Reconcile(ctx context.Context, n *runtimev1.Re
 					}
 					continue
 				}
-
+				paths, ok := parser.LocalDataToResourcepath[e.Path]
+				if ok {
+					changedPaths = append(changedPaths, keys(paths)...)
+					hasDuplicates = true
+				}
```

Comment:
Instead of leaking parser details here, I think this could be implemented inside the parser by adapting `parser.IsSkippable`, and adding the extra changed paths inside the parser's `parsePaths` function